LeaderboardBackend/Services/Impl/CategoryService.cs

```diff
@@ -29,9 +29,10 @@ public async Task<GetCategoriesForLeaderboardResult> GetCategoriesForLeaderboard
             return new NotFound();
         }
 
-        return board.Categories!
-            .Where(cat => includeDeleted || cat.DeletedAt == null)
-            .ToArray();
+        IEnumerable<Category> cats = board.Categories!.Where(cat => includeDeleted || cat.DeletedAt == null);
+        long count = cats.LongCount();
+        List<Category> items = cats.OrderBy(cat => cat.Id).Skip(page.Offset).Take(page.Limit).ToList();
```

Comment:
Maybe slap on the `Async` postfixes to maintain consistency with `LeaderboardService`:
```suggestion
        long count = await cats.LongCountAsync();
        
        // Ordering by ID is necessary, otherwise pagination breaks completely because the records won't
        // be returned in a specific order.
        
        List<Category> items = await cats.OrderBy(cat => cat.Id).Skip(page.Offset).Take(page.Limit).ToListAsync();
```

Reply:
Those methods do not exist on the `IEnumerable<T>` interface, but rather are extension methods for `IQueryable<T>`, which is a subtype (sub-interface?), so we don't have access to them here. This is because we actually already fetched all the categories from the DB and are just filtering them in memory. This would be something to fix in later PRs but it really doesn't matter too much here.